### ISSUE TITLE
_time is None when setting timezone through DateHelper

### ISSUE THREAD

**Issue by AaronCarton** [influxdata/influxdb-client-python#454]
__Steps to reproduce:__

1) set timezone as seen in: 
https://github.com/influxdata/influxdb-client-python/pull/238
2) query something similar as below
3) returned tables have None as start/stop/time fields

```py
date_utils.date_helper = DateHelper(timezone=tzlocal())
with InfluxDBClient(url=URL, token=TOKEN, org=ORG, timeout=25_000) as client:
   query= f'''import "date"
                   from(bucket: "{BUCKET}")
                      |> range(start: date.truncate(t: now(), unit: 1d), stop: now())
                      |> filter(fn: (r) => r._measurement == "Duiktank")
                      |> aggregateWindow(every: 1h, fn: sum)
                '''
   tables = client.query_api().query(query, org=ORG)
```

__Expected behavior:__
Query returns localized time/start/stop

__Actual behavior:__
Query returns None time/start/stop
![image](https://user-images.githubusercontent.com/78962840/173338498-cd148cde-b9f8-48d3-8ef8-c167cd2a532c.png)

__Specifications:__
 - Client Version: 1.29.1
 - InfluxDB Version: 
 - Platform: Windows

**Comment by bednar** [comment]
Hi @AaronCarton,

thanks for using our client.

> **Steps to reproduce:**
> 
> 1. set timezone as seen in:
>    [feat: add possibility to specify default `timezone` for datetimes without `tzinfo` #238](https://github.com/influxdata/influxdb-client-python/pull/238)
> 2. query something similar as below
> 3. returned tables have None as start/stop/time fields
> 
> ```python
> date_utils.date_helper = DateHelper(timezone=tzlocal())
> with InfluxDBClient(url=URL, token=TOKEN, org=ORG, timeout=25_000) as client:
>    query= f'''import "date"
>                    from(bucket: "{BUCKET}")
>                       |> range(start: date.truncate(t: now(), unit: 1d), stop: now())
>                       |> filter(fn: (r) => r._measurement == "Duiktank")
>                       |> aggregateWindow(every: 1h, fn: sum)
>                 '''
>    tables = client.query_api().query(query, org=ORG)
> ```
> 
> **Expected behavior:** Query returns localized time/start/stop
> 
> **Actual behavior:** Query returns None time/start/stop ![image](https://user-images.githubusercontent.com/78962840/173338498-cd148cde-b9f8-48d3-8ef8-c167cd2a532c.png)
> 
> **Specifications:**
> 
> * Client Version: 1.29.1
> * InfluxDB Version:
> * Platform: Windows

The #238 is supposed only for writes... I will update the `DateHelper` docs to clarify how to deal with your requirements. Meanwhile you can use following piece of code:


```python
from influxdb_client.client.util import date_utils
from influxdb_client.client.util.date_utils import DateHelper
import dateutil.parser
from dateutil import tz

def parse_date(date_string: str):
    return dateutil.parser.parse(date_string).astimezone(tzlocal())

date_utils.date_helper = DateHelper()
date_utils.date_helper.parse_date = parse_date
```

> Also worth noting that trying to use [Flux's timezone function](https://docs.influxdata.com/flux/v0.x/stdlib/timezone/location/) doesn't seem to work with this client either, as it throws an error stating `invalid import path timezone`

How looks your query? The following code works for me:

```python
p = {"_start": datetime.timedelta(hours=-1),
     "_location": "Prague",
     "_desc": True,
     "_floatParam": 25.1,
     "_every": datetime.timedelta(minutes=5)
     }

tables = query_api.query('''
    import "timezone"

    timezone.location(name: "America/Los_Angeles")
    
    from(bucket:"my-bucket") |> range(start: _start)
        |> filter(fn: (r) => r["_measurement"] == "my_measurement")
        |> filter(fn: (r) => r["_field"] == "temperature")
        |> filter(fn: (r) => r["location"] == _location and r["_value"] > _floatParam)
        |> aggregateWindow(every: _every, fn: mean, createEmpty: true)        
        |> sort(columns: ["_time"], desc: _desc) 
''', params=p)
```

Regards

**Comment by AaronCarton** [comment]
Thanks, this fixes part of the problem!

Although I'm now when I query data with `start: date.truncate(t: now(), unit: 1d)` it will return data started from 2 am instead of midnight (my timezone is 2 hours ahead of UTC)

I suspect this is because the `now()` portion still returns UTC instead of localized time? Which might be what https://docs.influxdata.com/flux/v0.x/stdlib/timezone/location/ is meant for?

**Comment by AaronCarton** [comment]
It seems that `import "timezone"` is only available on InfluxDB 2.1, and the server I'm querying is still running 2.0.8, I'll close this issue then 👍
